wrapper/src/main/java/software/amazon/jdbc/plugin/readwritesplitting/ReadWriteSplittingPlugin.java

```diff
@@ -506,10 +522,22 @@ private void getNewReaderConnection() throws SQLException {
     switchCurrentConnectionTo(this.readerConnection, this.readerHostSpec);
   }
 
+  private boolean isConnectionUsable(final CacheItem<Connection> cachedConnection) throws SQLException {
```

Comment:
Can you combine it with the following method?

Reply:
I don't think so without using templates, which doesn't really improve the performance.